### DISCUSSION TITLE
Update Row in Postgresql from Data Provider/Repeater Pattern

### DISCUSSION THREAD

**Discussion by tailorck** [Budibase/budibase#5739] [Help] [answered]
I am working on an application that requires the user to change the state of a row in the database for multiple columns sequentially. I have added buttons on a modal that will allow users to update these columns in the database. I am trying to write actions on these buttons to **_update_** the column for a particular row in the database, but it seems I can only add rows by using the `Save Row` action.

I see on the Budibase docs (https://docs.budibase.com/docs/actions) that it is possible to **_update_** the row, but it gives me the following error: `duplicate key value violates unique constraint "alert_pkey"`. 

<img width="1728" alt="image" src="https://user-images.githubusercontent.com/7613002/167063689-e7ebd139-d0a3-4485-85c6-b3280cb4e264.png">

How do I get around this error and update the row in the `alert` database with the new value for each column?

**Comment by tailorck** [comment]
Essentially, I want to take the row associated with `hw4(1).rtf` in the alerts table and update the `relevance` column to `true` when the user clicks the Relevant button. Is there a better way to do that? Currently, this page does not contain any forms.

**Comment by tailorck** [comment]
My latest solution is to create a new table in my database with the fields `case_id (pk)`, `alert_id (fk)`, `relevant`, `confirmed`. This works a bit better now because whenever the user clicks either `relevant` or `confirmed`, it creates a new entry in the `cases` table. However, since the user usually `confirms` a row after they mark it as `relevant`, the confirmation action still requires a row update rather than a new row save and the original problem persists.

How can I perform a row update on an existing table row in budibase?

**Comment by melohagan** [comment, marked as answer]
Hi @tailorck 

To perform an update on an existing row,  you will want to add a column for the **_id** and set this to match the existing id, so that a new one isn't generated.

So in this screenshot, the repeater is referring to a single row that has been filtered from an id passed from the URL into the data provider:

![Screenshot 2022-05-09 at 16 12 21](https://user-images.githubusercontent.com/101575380/167440833-750db89b-fe9b-4bf1-a6f7-e0917992ddc3.png)

More here: https://docs.budibase.com/docs/url-parameters#navigate-to-with-table-row-buttons

**Reply by melohagan** [reply]
Or you can use the repeater as the data source, and then you don't need to provide the id: https://docs.budibase.com/docs/actions#save-row

**Reply by tailorck** [reply]
Ok this did the trick! The primary key in my table was the `alert_id`. So I was trying to get budibase to recognize that I was updating a row by providing the `alert_id` as a column. What I did now was set the column to `_id` and the value to `{{ Repeater.alert_id }}` so that it updates the existing row rather than try to create a new one! Thank you!!!
